File: 프로그래머스/lv1/42840. 모의고사/모의고사.py
def solution(answers):
    list1 = [1, 2, 3, 4, 5]
    list2 = [2, 1, 2, 3, 2, 4, 2, 5]
    list3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    count = [0, 0, 0]
    
    for index, answer in enumerate(answers):
        if list1[index % len(list1)] == answer: count[0] += 1
        if list2[index % len(list2)] == answer: count[1] += 1
        if list3[index % len(list3)] == answer: count[2] += 1

    result = []
    max_result = max(count)
    for i, c in enumerate(count):
        if max_result == c:
            result.append(i + 1)
            
    return result

File: 프로그래머스/lv1/42840. 모의고사/모의고사.js
function solution(answers) {
  const score = [
    getScore(answers, [1, 2, 3, 4, 5]),
    getScore(answers, [2, 1, 2, 3, 2, 4, 2, 5]),
    getScore(answers, [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]),
  ];
  let answer = [];
  score.forEach((item, index) => {
    if (item === Math.max(...score)) {
      answer.push((index += 1));
    }
  });
  return answer;
}

function getScore(answers, numbers) {
  let score = 0;
  for (let i = 0; i < answers.length; i++) {
    let index = i % numbers.length;
    if (numbers[index] === answers[i]) score += 1;
  }
  return score;
}